catroid/src/androidTest/java/org/catrobat/catroid/test/embroidery/DSTFileGeneratorTest.java

```diff
@@ -0,0 +1,123 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.embroidery;
+
+import android.content.Context;
+import android.graphics.PointF;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.embroidery.DSTFileGenerator;
+import org.catrobat.catroid.embroidery.EmbroideryList;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class DSTFileGeneratorTest {
+	final String projectName = "testProject";
+	DSTFileGenerator fileGenerator;
+	EmbroideryList stitchpoints = new EmbroideryList();
+	File file;
+	Context context;
+	final int[] conversion = {-81, 81, -27, 27, -9, 9, -3, 3, -1, 1};
+
+	@Before
+	public void setUp() {
+		context = InstrumentationRegistry.getTargetContext().getApplicationContext();
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), projectName);
+		ProjectManager.getInstance().setCurrentProject(project);
+		fileGenerator = new DSTFileGenerator(context.getCacheDir());
+		file = new File(context.getCacheDir() + "/" + projectName + ".dst");
+	}
+
+	@Test
+	public void testConversionTable() {
+		for (int element = 0; element < DSTFileGenerator.CONVERSIONTABLE.length; element++) {
+			int mask = 0x200;
+			int value = 0;
+			for (int i = 0; i < conversion.length; i++) {
+				if ((DSTFileGenerator.CONVERSIONTABLE[element] & (mask >>> i)) != 0) {
+					value += conversion[i];
+				}
+			}
+			if (element > 121) {
+				assertEquals((element - 121) * (-1), value);
+			} else {
+				assertEquals(element, value);
+			}
+		}
+	}
+
+	@Test
+	public void testSimpleDSTFile() {
+		final int expectedFileLength = 521;
+		stitchpoints.clear();
+		stitchpoints.add(new PointF(1, 1));
+		stitchpoints.add(new PointF(2, 2));
+
+		fileGenerator.createDSTFile(stitchpoints);
+		long dstFileLength = file.length();
+
+		assertEquals(expectedFileLength, dstFileLength);
+	}
+
+	@Test
+	public void testStitchSplittingAllDirections() {
+		final int minExpectedFileLength = 542;
+
+		stitchpoints.clear();
+		stitchpoints.add(new PointF(0, 0));
+		stitchpoints.add(new PointF(250, 0));
+		stitchpoints.add(new PointF(250, 250));
+		stitchpoints.add(new PointF(0, 250));
+		stitchpoints.add(new PointF(0, 0));
+
+		fileGenerator.createDSTFile(stitchpoints);
+		long dstFileLength = file.length();
+
+		assertTrue(dstFileLength >= minExpectedFileLength);
```

Comment:
why ">="?
also asserting size is really minimalistic approach. have a comparison file in test res and compare it to the generated one would compare size and contents.